### PR TITLE
Make regex stricter and ES2015ify

### PR DESCRIPTION
**Pull request by kevva**
Also adds `xo`, updates `ava` and `download`. I just used the regex that was posted in https://github.com/sindresorhus/mapcode-regex/issues/3#issuecomment-217388948. Might be able to optimize it a little if we need to.

### CODE REVIEW

File: generate.js [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
-				ret.push(el['Full code']);
-			}
-		});
+		if (x['Full code'] !== '') {
+			ret.push(x['Full code']);
+		}
+	}
 
-		var out = '\'use strict\';\nmodule.exports = function () {\n\treturn /(?:(' + arrayUniq(ret).sort().join('|') + ') )?[A-Z0-9]{2,5}\.[A-Z0-9]{2,4}/g;\n};';
+	const out = `'use strict';\nmodule.exports = () => {\n\treturn /(?:(${arrayUniq(ret).sort().join('|')}) )?[ABCDEFGHJKLMNPQRSTUVWXYZabcdefghjklmnpqrstuvwxyz0-9]{2,}[.][ABCDEFGHJKLMNPQRSTUVWXYZabcdefghjklmnpqrstuvwxyz0-9]{2,}([-][0-9]{1,8})?/g;\n};\n`;
```
Comment (kevva) [inline generate.js]: Does `[.]` and `[-]` need to be inside those square brackets? Probably not.

File: generate.js [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
+	const data = files[0].data.toString();
+	const items = new CSV(data, {header: true}).parse();
+	const ret = [];
 
-		items.forEach(function (el) {
-			if (el.Territory !== '') {
-				ret.push(el.Territory);
-			}
+	for (const x of items) {
+		if (x.Territory !== '') {
```
Comment (kevva) [inline generate.js]: Any reason you check for `!== ''`? Noticed it was there since the first commit.

File: package.json [hunk 3]
```diff
@@ -10,10 +10,10 @@
     "url": "http://sindresorhus.com"
   },
   "engines": {
-    "node": ">=0.10.0"
+    "node": ">=4"
   },
   "scripts": {
-    "test": "node test.js",
+    "test": "xo && ava",
     "prepublish": "node generate.js"
```
Comment (sindresorhus) [inline package.json]: Can you rename `generate` to `make`?

File: generate.js [hunk 4]
```diff
@@ -1,31 +1,29 @@
...
-			if (el['Full code'] !== '') {
-				ret.push(el['Full code']);
-			}
-		});
+		if (x['Full code'] !== '') {
+			ret.push(x['Full code']);
+		}
+	}
 
-		var out = '\'use strict\';\nmodule.exports = function () {\n\treturn /(?:(' + arrayUniq(ret).sort().join('|') + ') )?[A-Z0-9]{2,5}\.[A-Z0-9]{2,4}/g;\n};';
```
Comment (sindresorhus) [inline generate.js:28]: You can use an inline arrow function here.

File: generate.js [hunk 5]
```diff
@@ -1,31 +1,29 @@
...
 
-new Download({extract: true})
-	.get('http://www.mapcode.com/kader/isotables.zip')
-	.run(function (err, files) {
-		var data = files[0].contents.toString();
-		var items = new CSV(data, {header: true}).parse();
-		var ret = [];
+download('http://www.mapcode.com/kader/isotables.zip', {extract: true}).then(files => {
+	const data = files[0].data.toString();
+	const items = new CSV(data, {header: true}).parse();
```
Comment (sindresorhus) [inline generate.js]: They now seem to ship a JS file too, with the data. We should rather use that.

File: generate.js (hunk 2 above)
Comment (sindresorhus) [reply generate.js]: Non-existent data are empty strings.

File: generate.js (hunk 1 above)
Comment (sindresorhus) [reply generate.js]: No, but `.` needs to be escaped.

File: test.js [hunk 6]
```diff
@@ -14,10 +13,10 @@ var fixture = [
 	'BR-AM 4J.Q2'
```
Comment (sindresorhus) [inline test.js]: Can you add a test for:

> high precision mapcodes are specified as XX.XX-123 (with 1-8 digits).

File: readme.md [hunk 7]
```diff
@@ -13,7 +13,7 @@ $ npm install --save mapcode-regex
 ## Usage
 
 ```js
-var mapcodeRegex = require('mapcode-regex');
+const mapcodeRegex = require('mapcode-regex');
 
 mapcodeRegex().test('4J.Q2');
 //=> true
```
Comment (sindresorhus) [inline readme.md:19]: Also update `http` to `https` in sindresorhus.com.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: generate.js (hunk 4 above)
Comment (kevva) [reply generate.js:28]: I tried doing that at first but it broke the module.

File: generate.js (hunk 5 above)
Comment (kevva) [reply generate.js]: None of the data is exported from the file though. It's just a `var` declaration.

Comment (sindresorhus) [reply generate.js]: I just saw that. Yeah, that's dumb. I guess it's easier to just stick with the CSV then.

Could you switch to use https://github.com/sindresorhus/neat-csv ?

File: generate.js (hunk 4 above)
Comment (sindresorhus) [reply generate.js:28]: Yeah, that's a V8 bug in Node.js 4. Just wrap it in parens.

File: generate.js (hunk 5 above)
Comment (kevva) [reply generate.js]: Yup, done.

**Review by sindresorhus — APPROVED** [review]

**Comment by sindresorhus** [comment]
Wee. Happy to finally being able to close #3 \o/ Thank you.

**Pull request merged**